lib/KBaseReport/utils/template_utils.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+
+from template import Template
+
+""" Utility for rendering a report from a template """
+
+def render_template_to_file(params):
+    """ Render a template and save the resulting content to a file
+
+    :param params:  (dict)  dict with keys
+
+        template_file:  # the template file to render
+        template_data:  # data to be rendered in the template
+        output_file:    # path to a file where the output will be written
+
+    :return:
+    { 'path': '/path/to/output_file' }
+
+    """
+    if ('template_file' not in params or 'output_file' not in params or 'template_data' not in params):
+        raise TypeError('Missing required parameters: please supply '
```

Comment:
`TypeError` usually applies to an incorrect type - a string that should've been a list for example. This would either be a ValueError (incorrect value - params is missing something) or, at a stretch, a KeyError.

Reply:
ValueError is it!

Reply:
Removed all this code as it was covered by the validator.